### PR TITLE
#60 Checking session and login decorator

### PR DESCRIPTION
**Pull request by bkuzmic**
For #60:

- created method to check session before each request and put user into request scope
- implemented login_required decorator - applied it for floors view

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#148](https://github.com/timelesslounge/timelessis/issues/148) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#148](https://github.com/timelesslounge/timelessis/issues/148) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/09d71dd3-957d-40be-aaa5-852fbcb94965); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/auth/views.py [hunk 1]
```diff
@@ -3,18 +3,36 @@
...
 )
+from timeless.employees.models import Employee
 
 bp = Blueprint("auth", __name__, url_prefix="/auth")
 
 
+@bp.before_app_request
+def load_logged_in_user():
+    user_id = session.get("user_id")
+    if user_id is None:
```
Comment (vladarefiev) [inline timeless/auth/views.py]: @bkuzmic I think you can just do `if not user_id:`

File: timeless/auth/views.py [hunk 2]
```diff
@@ -3,18 +3,36 @@
...
+        g.user = None
+    else:
+        g.user = Employee.query.get(user_id)
```
Comment (vladarefiev) [inline timeless/auth/views.py]: @bkuzmic What will be happened if `user_id` is blank line or for some reason Employee doesn't have this `id`?

File: timeless/auth/views.py [hunk 3]
```diff
@@ -3,18 +3,36 @@
...
+
+
+def login_required(view):
+    @wraps(view)
+    def wrapped_view(**kwargs):
+        if g.user is None:
```
Comment (vladarefiev) [inline timeless/auth/views.py]: @bkuzmic just `if not g.user:`

File: tests/test_floors.py [hunk 4]
```diff
@@ -1,12 +1,34 @@
...
+))
+def test_login_required(client, path):
+    response = client.post(path)
+    assert response.headers["Location"] == "http://localhost/auth/login"
+
+
+@pytest.mark.skip(reason="auth.login() is not yet implemented")
+def test_create(client, auth):
+    auth.login()
+    assert client.get("/floors/create").status_code == 200
```
Comment (vladarefiev) [inline tests/test_floors.py]: @bkuzmic in last PR @emilianodellacasa has asked to use `HTTPStatus.OK` from lib `from http import HTTPStatus`

File: tests/test_floors.py [hunk 5]
```diff
@@ -1,12 +1,34 @@
...
-def test_edit(client):
-    assert client.get("/floors/edit/1").status_code == 200     
+@pytest.mark.parametrize("path", (
+    "floors/create",
+    "/floors/edit/1",
+    "/floors/delete",
+))
+def test_login_required(client, path):
+    response = client.post(path)
+    assert response.headers["Location"] == "http://localhost/auth/login"
```
Comment (vladarefiev) [inline tests/test_floors.py]: @bkuzmic I think the best way is to use `url_for` here
http://flask.pocoo.org/docs/1.0/api/#flask.url_for

could we use it here?

**Comment by vladarefiev** [comment]
@bkuzmic looks good, a few comments

File: timeless/auth/views.py (hunk 2 above)
Comment (bkuzmic) [reply timeless/auth/views.py]: @vladarefiev It is fine. `Employee.query.get(user_id)` will return `None` for non-existing `user_id`

File: tests/test_floors.py (hunk 5 above)
Comment (bkuzmic) [reply tests/test_floors.py]: @vladarefiev Yes, we can. I'll do that.

**Comment by bkuzmic** [comment]
@vladarefiev Thanks for the comments. Please take a look again.

**Comment by vladarefiev** [comment]
@bkuzmic Thanks. Good for me

**Comment by vladarefiev** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 41 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#148](https://github.com/timelesslounge/timelessis/issues/148) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**